shell/platform/android/io/flutter/plugin/common/PluginRegistry.java

```diff
@@ -73,43 +77,68 @@
          * <p>When there is no foreground activity in the application, this
          * will return null. If a {@link Context} is needed, use context() to
          * get the application's context.</p>
+         *
+         * <p>This registrar is for Flutter's v1 embedding. To access an {@code Activity} from a
+         * plugin using the v2 embedding, see {@link ActivityPluginBinding#getActivity()}
```

Comment:
Though just reading the javadoc at ActivityPluginBinding.getActivity, it's not very clear how one acquires an activity plugin binding. I think ActivityPluginBinding still need a bit in its class javadoc on how to get the activity plugin binding. 

Also, I think we should probably point to flutter.dev/go/android-plugin-migration (which we'll fill in before 1.12) in all these JavaDocs for a holistic narrative.